codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -468,6 +464,9 @@ public boolean write(int allowedBytes) {
                             size == bytesWritten && endOfStream, writePromise);
                 } while (size != bytesWritten && allowedBytes > bytesWritten);
                 size -= bytesWritten;
+                if (size == 0 && endOfStream) {
+                    lifecycleManager.closeLocalSide(stream, promise);
```

Comment:
@buchgr - I think I like the `complete` method for this purpose.  Exit condition is either `complete` or `error`...easier to keep consistent across different frame types (thinking ahead to extension frames).  I think the issue was just we should make sure either `complete` or `error` gets called 1 time if possible. @nmittler - Am I missing something?